### ISSUE TITLE
Reduce the dependencies list for a new project

### ISSUE THREAD

**Issue by welingtonsampaio** [alboompro/boom-components#4]
Hi @all I saw an big list of peer dependencies to initiate a new project.

Today the list is: 

- classnames@^2.2.6
- formik@^1.5.7
- prop-types@^15.7.2
- react@^16.8.6
- react-component-component@^1.2.1
- react-dom@^16.8.6
- react-is@^16.8.6
- react-portal@^4.2.0
- shallowequal@^1.1.0
- styled-components@^4.2.0

I propose two solutions... 

1. if the dependency is small we can transform it into a dependency of the project;
2. analyze whether we really need these dependencies;

What you think guys?

**Comment by elisoncampos** [comment]
Those are the ones who I think are valid peer dependencies:

- react
- react-dom

Since we'll most likely have them listed as dependencies for the project that utilizes boom-components.

On the other hand:

- formik
- prop-types
-classnames
-styled-components
- react-portal

IMO, those should be dependencies, as for the time being, boom-components doesn't work without them. Maybe they're listed wrong on package.json?

For the other ones: react-is, react-component-component and shallowequal I agree with you. Maybe we can either have a in-house solution for their implementation or make them dependencies aswell.

**Comment by welingtonsampaio** [comment]
I agree with just about everything. I not only sure about the `react-portal`, because this dependency is necessary for older versions of react. In my opinion should be check if exists native portal in react if doesn't exists will import the `react-portal` with require, I don't know..

What do you think @elisoncampos?

**Comment by elisoncampos** [comment]
In any case, unless there's support for loading libraries outside package.json (like loading .mjs files works in chrome), there's no way around it. It'll be a dependency.

We could always drop support, since we use react-portal for our product and set react 16 as the start point. What you guys think?

**Comment by welingtonsampaio** [comment]
Maybe something like:

Importing...
```js
try {
    var Portal = require('react-portal');
} catch (ex) {
    var createPortal = require('react').createPortal;
}
```

Using...
```js
class El {
  render() {
    if (Portal) {
      return (
        <Portal>content...</Portal>
      );
    } else {
      return createPortal(
        <div>content...</div>,
        document.body
      );
    }
  }
}
```

**Comment by elisoncampos** [comment]
Wouldn't that be the opposite?

```js
try {
  var Portal = React.createPortal();
} catch (_e) {
  var Portal = require("react-portal");
}
```

Check first for react implementation of portal and if failing, use dependency fallback? Also, in this case, react-portal should be a dependency as having it a peer-dependency would lead us to inform the need for projects using older react having to install it to work (which is not a bad approach per se).

**Comment by welingtonsampaio** [comment]
I agree with everything! Because we strongly recommend using React16 +, but we can use it in older versions.
